Makefile

```diff
@@ -306,15 +306,26 @@ docker-images-jaeger-backend-debug: SUFFIX = -debug
 
 .PHONY: docker-images-jaeger-backend docker-images-jaeger-backend-debug
 docker-images-jaeger-backend docker-images-jaeger-backend-debug: create-baseimg create-debugimg
-	for component in agent collector query ingester ; do \
-		docker build --target $(TARGET) \
-			--tag $(DOCKER_NAMESPACE)/jaeger-$$component$(SUFFIX):${DOCKER_TAG} \
+	# Save the current Internal Field Separator (IFS) for later recovery, then set it to a comma-separator.
+	# The reason for using a comma-separator is to support the empty prefix required for all-in-one.
+	OLDIFS=$IFS; \
+	IFS=","; \
+	for pair in "jaeger,agent" "jaeger,collector" "jaeger,query" "jaeger,ingester" ",all-in-one" ; do \
```

Comment:
Open to other suggestions on a better approach.

Reply:
perhaps it's simpler to use strings like `jaeger-agent` and then just regex-out the `jaeger-` prefix when referring to the working dir `cmd/$$component`

Reply:
Good suggestion, done in: [353cc39](https://github.com/jaegertracing/jaeger/pull/3813/commits/353cc392292a3284dce8662edddeb747155866a4)